src/Nethermind/Nethermind.State/IStorageProvider.cs

```diff
@@ -18,14 +18,18 @@
 
 namespace Nethermind.State
 {
-    public interface IStorageProvider : IJournal<int>
+    public interface IStorageProvider : IJournal<Snapshot>
```

Comment:
Maybe lets create some specialized snapshot, like inner Snapshot.Storage struct? Now its confusing as  there is State there too.

Reply:
Good idea. Let me know if that's what you had in mind.